manifests/charts/ztunnel/templates/daemonset.yaml

```diff
@@ -87,6 +92,10 @@ spec:
           value: {{ .Values.logLevel | quote }}
         - name: ISTIO_META_CLUSTER_ID
           value: {{ .Values.multiCluster.clusterName | default "Kubernetes" }}
+        - name: INPOD_ENABLED
+          value: "true"
+        - name: ISTIO_META_DNS_PROXY_ADDR
+          value: "127.0.0.1:15053"
```

Comment:
might need add  `ISTIO_META_DNS_CAPTURE｀　 as dns_proxy will only be created in ztunnel as in https://github.com/istio/ztunnel/blob/2d4effb1076898a5675a95f525d0cdfaedd9d6fc/src/app.rs#L180

Reply:
Yes, that is required for capture, but as mentioned here: https://github.com/istio/istio/pull/48253#discussion_r1466716667 we do not want to enable by default.

It needs to be enabled in 2 places (ztunnel to enable DNS proxying in the first place, with `ISTIO_META_DNS_CAPTURE` and then CNI as well, to enable redirection. 

The CNI value isn't plumbed thru to the config, so fixing that.

Reply:
https://github.com/istio/istio/pull/48253/commits/277db3d0b60b860343bc3875710e26d3772b7143 should do it.

Unfortunately, because we want it to be optional, it effectively requires 2 knobs for everything to work.

- One to enable DNS capture in CNI
- One to enable the DNS proxy server in ztunnel

Reply:
We can probably consolidate this to be the same knob.

Reply:
+1 since this PR is already big enough